src/metabase/api/database.clj

```diff
@@ -551,10 +551,11 @@
              :%lower.metabase_field/name     [:like (u/lower-case-en search-string)]
              :metabase_field.visibility_type [:not-in ["sensitive" "retired"]]
              :table.db_id                    db-id
-             {:order-by  [[[:lower :metabase_field.name] :asc]
-                          [[:lower :table.name] :asc]]
-              :left-join [[:metabase_table :table] [:= :table.id :metabase_field.table_id]]
-              :limit     limit}))
+             {:order-by   [[[:lower :metabase_field.name] :asc]
+                           [[:lower :table.name] :asc]]
+              :inner-join [[:metabase_table :table] [:and :table.active
```

Comment:
It might be worth commenting that it was (counter-intuitively) faster to have the "active" check within the join, in case someone comes to optimize again.

Reply:
Hah, it felt intuitive for me - you get less data to join this way 😁 But sure, I can leave a comment there.

Reply:
Probably you're right, my intuition is quite rusty and I didn't give much thought to it 😆